### PR TITLE
Code Quality: Improved UserControls codebase

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

This namespace was one of the places where there were leftovers from UWP.

Re-organized codebase in UserControls to enhance maintainability and removed some XAML files(OpacityIcon and DataGridHeader), so it must be little but the app should be faster a bit.

## Task Checklist

- [x] Rename `IAddressToolBar` to `IAddressToolbarViewModel`
- [x] Rename `ISearchBox` to `ISearchBoxViewModel`
- [x] Rename `ISearchBox` to `ISearchBoxViewModel`
- [x] Rename `Views.LayoutModes` to `Views.ContentLayouts`
- [x] Rename `ViewModels.LayoutModes` to `ViewModels.ContentLayouts`
- [x] Move `IAddressToolbarViewModel` from `UserControls` to `ViewModels.UserControls`
- [x] Move `ISearchBoxViewModel` from `UserControls` to `ViewModels.UserControls`
- [x] Move `ViewModels.Previews` to `ViewModel.UserControls.Previews`
- [x] Move `ViewModels.Widgets` to `ViewModel.UserControls.Widgets`
- [x] Change base class of `DataGridHeader` from `UserControl` to `ButtonBase`
- [x] Ensure we folllow our codebase guideline
- [x] Clean up
- [ ] Apply requested changes

## Known Issues

Following issues would be fixed in the upcoming PR.

_N/A_

## PR Checklist

- [x] **Related**: #4180
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Steps To Validate Changes

There're some classes that were changed codebase largely. Please have look at:

- See that DataGridHeader is working well
- See that OpacityIcon is working well

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do, instead.

### CODE REVIEW

**Comment by 0x5bfa** [comment]
This is ready for review.

**Review by QuaintMako — COMMENTED** [review]
Small feedback, I cannot test it.
LGTM overall.

File: src/Files.App/UserControls/AddressToolbar.xaml.cs [hunk 1]
```diff
@@ -81,10 +100,11 @@ private void ManualPathEntryItem_Click(object _, PointerRoutedEventArgs e)
 		{
 			if (e.Pointer.PointerDeviceType is PointerDeviceType.Mouse)
 			{
-				var ptrPt = e.GetCurrentPoint(NavToolbar);
+				var ptrPt = e.GetCurrentPoint(this);
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/UserControls/AddressToolbar.xaml.cs:103]: there was x:Name="NavToolbar" in this AddressToolbar class. but means the same.

File: src/Files.App/UserControls/DataGridHeader/DataGridHeader.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		protected override void OnPointerReleased(PointerRoutedEventArgs e)
+		{
+			base.OnPointerReleased(e);
+			VisualStateManager.GoToState(this, NormalState, true);
+		}
+
+		//protected override AutomationPeer OnCreateAutomationPeer()
+		//{
+		//	return new DataGridHeaderAutomationPeer(this);
+		//}
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/UserControls/DataGridHeader/DataGridHeader.cs:163]: To remind devs that this control should have AutomationPeer. I will add a TODO.

File: src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs [hunk 3]
```diff
@@ -1,64 +1,72 @@
...
 namespace Files.App.UserControls
 {
 	public sealed partial class InnerNavigationToolbar : UserControl
 	{
-		public InnerNavigationToolbar()
-		{
-			InitializeComponent();
-			PreviewPaneViewModel = Ioc.Default.GetRequiredService<PreviewPaneViewModel>();
-		}
+		private readonly IAddItemService _addItemService = Ioc.Default.GetRequiredService<IAddItemService>();
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: How come?

[Runtime Code Style Conventions](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) Itme 3:

> We use _camelCase for internal and private fields and use readonly where possible.

[Our Code Stlye Guideline](https://files.community/docs/contributing/code-style) Section 2, Item 4:

> Use camelCase prefixed with _ for private fields

[the remark this answers is not in the document]
Comment (QuaintMako) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: In that case, the two services below are not named properly.

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: Yeah, I agree with you. I will fix.

But as my policy ones that are referenced from XAML should be named with PascalCase; Commands & ModifiableCommands because when accessing in XAML, the code will be like as follows if CommandManager is defines in code-behind.
```xaml
<Button Command="Commands.OpenInNewTab" />
```

Is that not bad practice? I wanna hear from you.

[the remark this answers is not in the document]
Comment (QuaintMako) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: There is something off with the second format indeed, it looks really off. Yet at the same time, if we're making an exception to our coding rules, than I believe the coding rules are not good enough.

On a first read, one can be confused as to why some private fields are marked with an underscore and why others are not. 

Maybe we can mark the fields used in xmal with a letter (bad idea in my opinion). Maybe we can add a rule in the coding rules about having them capitalized as a way to mark their use in XMAL (will work for a while, but give it a year or two and the whole thing gonna be a real mess).

I need to think about it a little bit to see if anything clever can be made. Maybe we can find some inputs online for good practice.

Either way, that is something minor here. We can open an issue about that to keep track of it.

Tagging @d2dyno1 since I believe he wrote the coding rules, he might have some good input about it?

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: Making them properties would fix issues. How about?

[the remark this answers is not in the document]
Comment (d2dyno1) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml.cs:14]: Why not just use a private property in this case?

**Comment by yaira2** [comment]
I'm not so sure about renaming `LayoutModes` to `ContentLayouts`, can you clarify where this is coming from?

**Comment by yaira2** [comment]
Which steps were used to verify this changes?

File: src/Files.App/App.xaml [hunk 4]
```diff
@@ -30,9 +30,9 @@
 				<XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
 
 				<!--  Styles for the custom icons  -->
-				<ResourceDictionary Source="/ResourceDictionaries/PathIcons.xaml" />
-				<ResourceDictionary Source="/UserControls/SideBar/SideBarControls.xaml" />
-				<ResourceDictionary Source="ms-appx:///ResourceDictionaries/App.Theme.TextBlockStyles.xaml" />
+				<ResourceDictionary Source="ms-appx:///ResourceDictionaries/PathIcons.xaml" />
+				<ResourceDictionary Source="ms-appx:///UserControls/SideBar/SideBarControls.xaml" />
+				<ResourceDictionary Source="ms-appx:///ResourceDictionaries/TextBlockStyles.xaml" />
```
Comment (yaira2) [inline src/Files.App/App.xaml:35]:
```suggestion
				<ResourceDictionary Source="ms-appx:///ResourceDictionaries/App.Theme.TextBlockStyles.xaml" />
```
The `.` makes the code more readable.

File: src/Files.App/ResourceDictionaries/TextBlockStyles.xaml
Comment (yaira2) [inline src/Files.App/ResourceDictionaries/TextBlockStyles.xaml:1]: Can you revert this change?

File: src/Files.App/Views/ContentLayouts/DetailsLayoutBrowser.xaml [hunk 5]
```diff
@@ -149,7 +150,7 @@
 				</Setter>
 			</Style>
 
-			<Style TargetType="uc:DataGridHeader">
+			<Style BasedOn="{StaticResource DefaultDataGridHeaderStyle}" TargetType="uc:DataGridHeader">
```
Comment (yaira2) [inline src/Files.App/Views/ContentLayouts/DetailsLayoutBrowser.xaml:153]: It might be better to do this in a separate PR so we can review the change on it's own.
